aten/src/ATen/native/ReduceOpsUtils.h

```diff
@@ -37,6 +37,10 @@ static bool _dimreduce_return_trivial_no_ident(Tensor &result, const Tensor &sel
     return true;
   }
 
+  if (self.numel() == 0 && self.ndimension() > 0) {
```

Comment:
this can't be correct, because a tensor with `numel == 0` can't have have 0 dimensions (all 0-dimensional tensors have 1 element).

Reply:
I actually didn't understand your comment in #17750. I thought you wanted to replicate numpy's behavior, and that's what I did. This condition was added to allow `numel ==0` tensors to have dimensions, again, to match numpy's behavior.